src/Models/ShallowWaterModels/solution_and_tracer_tendencies.jl

```diff
@@ -78,5 +78,12 @@ end
 
     @inbounds c = tracers[tracer_index]
 
-    return ( 0.0 )
+    u = solution.uh / ℑxyᶠᶠᵃ(i, j, k, grid, solution.h)   # are these computed correctly?
+    v = solution.vh / ℑxyᶠᶠᵃ(i, j, k, grid, solution.h)
+
+    return ( 0.0
+             - div_ucvc(i, j, k, grid, u, v, c)           # are these evaluated at the correct locations?
+             + c * ∂xᶜᵃᵃ(i, j, k, grid, u)
+             + c * ∂yᵃᶜᵃ(i, j, k, grid, v) 
+            )
```

Comment:
I think you're mixing fields and `field[i, j, k]`?

I'm guessing you might want something similar to https://github.com/CliMA/Oceananigans.jl/blob/a211d9d0f0ddf3f66ca9fe4319f47aac1e9e6164/src/Advection/tracer_advection_operators.jl but for shallow water (might have to define `advective_tracer_flux_x` and `advective_tracer_flux_y` for shallow water)? Might end up similar to what you did for shallow water momentum advection operators above.

Reply:
Yes, I am certainly mixed up.  Thanks for sharing the link.  I tried to copy that before but will try it again and see if it works any better the next time around.